### ISSUE TITLE
openssh-sftp-clien 0.14 seems broken

### ISSUE THREAD

**Issue by Xuanwo** [openssh-rust/openssh-sftp-client#111]
https://github.com/apache/incubator-opendal/issues/3466#issuecomment-1801183456

```shell
   Compiling concurrent_arena v0.1.8
   Compiling openssh-sftp-protocol v0.24.0
   Compiling openssh-sftp-error v0.3.1
   Compiling openssh-sftp-client-lowlevel v0.5.1
   Compiling openssh-sftp-client v0.14.0
error[E0277]: the trait bound `openssh_sftp_client_lowlevel::Error: From<openssh::Error>` is not satisfied
  --> /home/xuanwo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openssh-sftp-client-0.14.0/src/sftp/openssh_session.rs:65:30
   |
65 |         Err(err) => Some(err.into()),
   |                              ^^^^ the trait `From<openssh::Error>` is not implemented for `openssh_sftp_client_lowlevel::Error`
   |
   = help: the following other types implement trait `From<T>`:
             <openssh_sftp_client_lowlevel::Error as From<openssh_sftp_error::AwaitableError>>
             <openssh_sftp_client_lowlevel::Error as From<SshFormatError>>
             <openssh_sftp_client_lowlevel::Error as From<openssh::error::Error>>
             <openssh_sftp_client_lowlevel::Error as From<std::io::Error>>
             <openssh_sftp_client_lowlevel::Error as From<JoinError>>
             <openssh_sftp_client_lowlevel::Error as From<TryFromIntError>>
   = note: required for `openssh::Error` to implement `Into<openssh_sftp_client_lowlevel::Error>`

error[E0277]: the trait bound `openssh_sftp_client_lowlevel::Error: From<openssh::Error>` is not satisfied
  --> /home/xuanwo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openssh-sftp-client-0.14.0/src/sftp/openssh_session.rs:76:26
   |
76 |     let occuring_error = session.close().await.err().map(Error::from);
   |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<openssh::Error>` is not implemented for `openssh_sftp_client_lowlevel::Error`
   |
   = help: the following other types implement trait `From<T>`:
             <openssh_sftp_client_lowlevel::Error as From<openssh_sftp_error::AwaitableError>>
             <openssh_sftp_client_lowlevel::Error as From<SshFormatError>>
             <openssh_sftp_client_lowlevel::Error as From<openssh::error::Error>>
             <openssh_sftp_client_lowlevel::Error as From<std::io::Error>>
             <openssh_sftp_client_lowlevel::Error as From<JoinError>>
             <openssh_sftp_client_lowlevel::Error as From<TryFromIntError>>

error[E0277]: the trait bound `openssh_sftp_client_lowlevel::Error: From<openssh::Error>` is not satisfied
  --> /home/xuanwo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openssh-sftp-client-0.14.0/src/sftp/openssh_session.rs:76:58
   |
76 |     let occuring_error = session.close().await.err().map(Error::from);
   |                                                          ^^^^^^^^^^^ the trait `From<openssh::Error>` is not implemented for `openssh_sftp_client_lowlevel::Error`
   |
   = help: the following other types implement trait `From<T>`:
             <openssh_sftp_client_lowlevel::Error as From<openssh_sftp_error::AwaitableError>>
             <openssh_sftp_client_lowlevel::Error as From<SshFormatError>>
             <openssh_sftp_client_lowlevel::Error as From<openssh::error::Error>>
             <openssh_sftp_client_lowlevel::Error as From<std::io::Error>>
             <openssh_sftp_client_lowlevel::Error as From<JoinError>>
             <openssh_sftp_client_lowlevel::Error as From<TryFromIntError>>

error: cannot check whether the hidden type of opaque type satisfies auto traits
   --> /home/xuanwo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openssh-sftp-client-0.14.0/src/sftp/openssh_session.rs:105:35
    |
105 |         let handle = tokio::spawn(create_session_task(session, tx));
    |                      ------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                      |
    |                      required by a bound introduced by this call
    |
note: opaque type is declared here
   --> /home/xuanwo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openssh-sftp-client-0.14.0/src/sftp/openssh_session.rs:25:6
    |
25  | ) -> Option<Error> {
    |      ^^^^^^^^^^^^^
note: this item depends on auto traits of the hidden type, but may also be registering the hidden type. This is not supported right now. You can try moving the opaque type and the item that actually registers a hidden type into a new submodule
   --> /home/xuanwo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openssh-sftp-client-0.14.0/src/sftp/openssh_session.rs:99:18
    |
99  |     pub async fn from_session(
    |                  ^^^^^^^^^^^^
note: required by a bound in `tokio::spawn`
   --> /home/xuanwo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.33.0/src/task/spawn.rs:166:21
    |
164 |     pub fn spawn<F>(future: F) -> JoinHandle<F::Output>
    |            ----- required by a bound in this function
165 |     where
166 |         F: Future + Send + 'static,
    |                     ^^^^ required by this bound in `spawn`

error[E0277]: `?` couldn't convert the error to `openssh_sftp_client_lowlevel::Error`
   --> /home/xuanwo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openssh-sftp-client-0.14.0/src/sftp/openssh_session.rs:110:27
    |
110 |             Ok(res) => res?,
    |                           ^ the trait `From<openssh::Error>` is not implemented for `openssh_sftp_client_lowlevel::Error`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = help: the following other types implement trait `From<T>`:
              <openssh_sftp_client_lowlevel::Error as From<openssh_sftp_error::AwaitableError>>
              <openssh_sftp_client_lowlevel::Error as From<SshFormatError>>
              <openssh_sftp_client_lowlevel::Error as From<openssh::error::Error>>
              <openssh_sftp_client_lowlevel::Error as From<std::io::Error>>
              <openssh_sftp_client_lowlevel::Error as From<JoinError>>
              <openssh_sftp_client_lowlevel::Error as From<TryFromIntError>>
    = note: required for `Result<Sftp, openssh_sftp_client_lowlevel::Error>` to implement `FromResidual<Result<Infallible, openssh::Error>>`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `openssh-sftp-client` (lib) due to 5 previous errors
cargo build  22.54s user 3.50s system 251% cpu 10.364 total
```

**Comment by Xuanwo** [comment]
cc @NobodyXu, would you like to take a look?

**Comment by NobodyXu** [comment]
Thanks, confirmed that I can reproduce this by adding:

```rust
[dependencies]
openssh-sftp-client = { version = "0.14.0", features = ["tracing", "openssh"]}
```

to a new project, however I cannot produce this by running `cargo b --all-features` in `openssh-sftp-client`.

**Comment by NobodyXu** [comment]
Aha I found the problem:

```
    │   ├── openssh-sftp-error v0.3.1
    │   │   ├── awaitable-error v0.1.0
    │   │   ├── openssh v0.9.9
    │   │   │   ├── dirs v4.0.0
    │   │   │   │   └── dirs-sys v0.3.7
    │   │   │   │       └── libc v0.2.150
    │   │   │   ├── libc v0.2.150
    │   │   │   ├── once_cell v1.18.0
    │   │   │   ├── shell-escape v0.1.5
    │   │   │   ├── tempfile v3.8.1 (*)
    │   │   │   ├── thiserror v1.0.50 (*)
    │   │   │   ├── tokio v1.33.0 (*)
    │   │   │   └── tokio-pipe v0.2.12 (*)
    │   │   ├── openssh-sftp-protocol-error v0.1.0
    │   │   │   ├── serde v1.0.192 (*)
    │   │   │   ├── thiserror v1.0.50 (*)
    │   │   │   └── vec-strings v0.4.8
    │   │   │       ├── serde v1.0.192 (*)
    │   │   │       └── thin-vec v0.2.12
    │   │   ├── ssh_format_error v0.1.0
    │   │   │   └── serde v1.0.192 (*)
    │   │   ├── thiserror v1.0.50 (*)
    │   │   └── tokio v1.33.0 (*)
```

`openssh-sftp-error` is still using `openssh`, I forgot to release a new version.

**Comment by NobodyXu** [comment]
I've released openssh-sftp-error v0.3.2, after running `cargo update` to update `openssh-sftp-error` to latest version, the compilation error is fixed.

**Comment by sd44** [comment]
Quick fix nad more constructive reply.
